spec/cli/test/builder-cache.spec.js

```diff
@@ -0,0 +1,114 @@
+// MIT License
+//
+// Copyright 2016 Electric Imp
+//
+// SPDX-License-Identifier: MIT
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+// Builder syntax tests
+
+'use strict';
+
+require('jasmine-expect');
+const fs = require('fs');
+const run = require('./run');
+
+describe('TestCommand test suite for Builder cache scenario', () => {
+  let commandOut = '',
+    commandSuccess = true;
+
+  beforeEach(() => {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 100000;
+  });
+
+  // reset previous state
+  var removeCacheDir = function(path) {
+    if (fs.existsSync(path)) {
+      fs.readdirSync(path).forEach(function(file, index){
+        var cachedLib = path + "/" + file;
+        if (fs.lstatSync(cachedLib).isDirectory()) { // unexpected case
+          removeCacheDir(cachedLib);
+        } else // remove lib
+          fs.unlinkSync(cachedLib);
+      });
+      fs.rmdirSync(path);
+    }
+  };
+  removeCacheDir(".builder-cache");
```

Comment:
shouldn't it be called from `beforeEach`?

Reply:
There is only one case which create a real cache folder

Reply:
I have tried to add it into to the "beforeeach" but command and verification are separated and it lead to file remove folder before check and as a result test failure